src/contentProvider.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { mavenExplorerProvider } from "./explorer/mavenExplorerProvider";
+import { IEffectivePom } from "./explorer/model/IEffectivePom";
+import { MavenProject } from "./explorer/model/MavenProject";
+import { getDependencyTree } from "./handlers/showDependenciesHandler";
+import { Utils } from "./utils/Utils";
+
+class MavenContentProvider implements vscode.TextDocumentContentProvider {
+    public async provideTextDocumentContent(uri: vscode.Uri, _token: vscode.CancellationToken): Promise<string | undefined> {
+        if (uri.scheme !== "maven") {
```

Comment:
Consider use scheme like `vscode-maven` to specify this is a vscode context based scheme?

Just like the VS Code team does for the virtual file system, which is called `vscode-vfs`.